### PR TITLE
refactor: create use case for account deletion

### PR DESCRIPTION
**Pull request by Polpetta**
This won't change the behavior but it would:
* add `shellspec` as a acceptance test functionality
* create new package `usecase` to start reasoning with onion/hexagon structure

Ref: CO-731

### CODE REVIEW

File: store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+import com.zimbra.common.util.Log;
+import com.zimbra.common.util.ZimbraLog;
+import com.zimbra.cs.account.Provisioning;
+import com.zimbra.cs.mailbox.MailboxManager;
+import io.vavr.control.Try;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.inject.Named;
+
+public class DeleteUserUseCase {
```
Comment (keshavbhatt) [inline store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java:13]: Consider adding javadoc.

File: store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  public DeleteUserUseCase(
+      @Named("defaultProvisioning") Provisioning provisioning,
+      @Named("defaultMailboxManager") MailboxManager mailboxManager,
+      @Named("zimbraLogSecurity") Log log) {
+    this.provisioning = provisioning;
+    this.mailboxManager = mailboxManager;
+    this.log = log;
+  }
+
+  public Try<Void> delete(String userId) {
```
Comment (keshavbhatt) [inline store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java:29]: Consider adding javadoc.

**Review by keshavbhatt — APPROVED** [review]
Just asked for adding javadoc is classes and methods.

**Comment by keshavbhatt** [comment]
@Polpetta also the title of the PR doesn't seems to match the changes we are adding, consider renaming it accordingly.

File: store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java (hunk 1 above)
Comment (Polpetta) [reply store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java:19]: Done!

File: store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java (hunk 2 above)
Comment (Polpetta) [reply store/src/main/java/com/zextras/mailbox/domain/usecase/DeleteUserUseCase.java:42]: Done!

**Pull request merged**